src/test/java/seedu/address/testutil/EditStudentDescriptorBuilder.java

```diff
@@ -0,0 +1,88 @@
+package seedu.address.testutil;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.student.EditStudentCommand;
+import seedu.address.logic.commands.student.EditStudentCommand.EditStudentDescriptor;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Matric;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Student;
+import seedu.address.model.tag.Tag;
+
+/**
+ * A utility class to help with building EditStudentDescriptor objects.
+ */
+public class EditStudentDescriptorBuilder {
+
+    private EditStudentDescriptor descriptor;
+
+    public EditStudentDescriptorBuilder() {
+        descriptor = new EditStudentDescriptor();
+    }
+
+    public EditStudentDescriptorBuilder(EditStudentCommand.EditStudentDescriptor descriptor) {
+        this.descriptor = new EditStudentDescriptor(descriptor);
+    }
+
+    /**
+     * Returns an {@code EditStudentDescriptor} with fields containing {@code student}'s details
+     */
+    public EditStudentDescriptorBuilder(Student student) {
+        descriptor = new EditStudentCommand.EditStudentDescriptor();
```

Comment:
should just use `EditStudentDescriptor `directly instead of referring to `EditStudentCommand`